renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -1079,6 +1082,45 @@ VkResult WrappedVulkan::FilterDeviceExtensionProperties(VkPhysicalDevice physDev
 
   if(pLayerName == NULL)
   {
+    InstanceDeviceInfo *instDevInfo = GetRecord(m_Instance)->instDevInfo;
+
+    // extensions with conditional support
+    for(auto it = filtered.begin(); it != filtered.end();)
+    {
+      if(!strcmp(it->extensionName, VK_EXT_FRAGMENT_DENSITY_MAP_EXTENSION_NAME))
+      {
+        // require GPDP2
+        if(instDevInfo->ext_KHR_get_physical_device_properties2)
+        {
+          VkPhysicalDeviceFragmentDensityMapFeaturesEXT fragmentDensityFeatures = {
+              VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FRAGMENT_DENSITY_MAP_FEATURES_EXT};
+          VkPhysicalDeviceFeatures2 base = {VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2};
+          base.pNext = &fragmentDensityFeatures;
+          ObjDisp(physDev)->GetPhysicalDeviceFeatures2(Unwrap(physDev), &base);
+
+          if(fragmentDensityFeatures.fragmentDensityMapNonSubsampledImages)
+          {
+            // supported
```

Comment:
We also need to search for the `VkPhysicalDeviceFragmentDensityMapFeaturesEXT` in the `pNext` chain of `VkDeviceCreateInfo`, and if we find it then force on this feature as the application may not have. We can at least assume that if fragment density maps are used at all then the features struct will be in the chain (since it's needed to enable the general feature), we just need to poke this bool we care about.

Reply:
Added code in vk_device_funcs.cpp that should fix this, and verified it gets called when I add a `VkPhysicalDeviceFragmentDensityMapFeaturesEXT with fragmentDensityMapNonSubsampledImages = false